src/context/scope_map.rs

```diff
@@ -5,22 +5,20 @@
 //! before the current one, until it finds the correct component.
 
 use std::collections::{HashMap, LinkedList};
-use std::rc::Rc;
 
-use crate::instruction::{FunctionDec, TypeDec, Var};
-use crate::{ErrKind, Error, Instruction};
+use crate::{ErrKind, Error};
 
-/// A scope contains a set of available variables and functions
+/// A scope contains a set of available variables, functions and types
 #[derive(Clone)]
-struct Scope {
-    variables: HashMap<String, Var>,
-    functions: HashMap<String, Rc<FunctionDec>>,
-    types: HashMap<String, Rc<TypeDec>>,
+pub struct Scope<V, F, T> {
+    pub(crate) variables: HashMap<String, V>,
+    pub(crate) functions: HashMap<String, F>,
+    pub(crate) types: HashMap<String, T>,
```

Comment:
When are these fields accessed from outside ?

Reply:
When implementing the `Display` trait in the `Context`